nbconvert/exporters/templateexporter.py

```diff
@@ -560,18 +560,37 @@ def get_template_names(self):
                 template_dir = os.path.join(base_dir, base_template)
                 if os.path.exists(template_dir):
                     found_at_least_one = True
-                conf_file = os.path.join(template_dir, 'conf.json')
-                if os.path.exists(conf_file):
-                    with open(conf_file) as f:
-                        conf = recursive_update(json.load(f), conf)
-            for root_dir in root_dirs:
-                template_dir = os.path.join(root_dir, 'nbconvert', 'templates', base_template)
-                if os.path.exists(template_dir):
-                    found_at_least_one = True
-                conf_file = os.path.join(template_dir, 'conf.json')
-                if os.path.exists(conf_file):
-                    with open(conf_file) as f:
-                        conf = recursive_update(json.load(f), conf)
+                    conf_file = os.path.join(template_dir, 'conf.json')
+                    if os.path.exists(conf_file):
+                        with open(conf_file) as f:
+                            conf = recursive_update(json.load(f), conf)
+                    break
+            if not found_at_least_one:
+                for root_dir in root_dirs:
+                    template_dir = os.path.join(root_dir, 'nbconvert', 'templates', base_template)
+                    if os.path.exists(template_dir):
+                        found_at_least_one = True
+                        conf_file = os.path.join(template_dir, 'conf.json')
+                        if os.path.exists(conf_file):
+                            with open(conf_file) as f:
+                                conf = recursive_update(json.load(f), conf)
+                        break
```

Comment:
Same here. Not sure the recursion on root_dirs should be conditional on `found_at_least_one`.

Reply:
The recursion isn't conditional, it recurses to completion before returning back to this context